packages/payment-detection/src/eth/input-data.ts

```diff
@@ -8,10 +8,37 @@ import Utils from '@requestnetwork/utils';
 import PaymentReferenceCalculator from '../payment-reference-calculator';
 
 import EthInputDataInfoRetriever from './info-retriever';
+import EthProxyInputDataInfoRetriever from './proxy-info-retriever';
 
 const bigNumber: any = require('bn.js');
 const supportedNetworks = ['mainnet', 'rinkeby', 'private'];
 
+interface IProxyContractByVersionByNetwork {
+  [version: string]: {
+    [network: string]: { address: string; creationBlockNumber: number };
+  };
+}
+
+const firstVersion = {
```

Comment:
Add comment.
The name is really not clear, I have no idea what this is from the name.